src/test/java/api/loans/CheckInByBarcodeTests.java

```diff
@@ -1038,6 +1041,39 @@ public void overdueFineCalculatedCorrectlyWhenHourlyFeeFinePolicyIsApplied() {
       homeLocation.getJson().getString("name"), ownerId, feeFineId, 7.0));
   }
 
+  @Test
+  public void itemCheckedInEventIsPublished() {
+    final IndividualResource james = usersFixture.james();
+    final UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+    final IndividualResource homeLocation = locationsFixture.basedUponExampleLocation(
+      item -> item.withPrimaryServicePoint(checkInServicePointId));
+    final IndividualResource nod = itemsFixture.basedUponNod(item ->
+      item.withPermanentLocation(homeLocation.getId()));
+
+    DateTime checkOutDate = new DateTime(2020, 1, 18, 18, 0, 0, UTC);
+    DateTime checkInDate = new DateTime(2020, 1, 22, 15, 30, 0, UTC);
+
+    checkOutFixture.checkOutByBarcode(nod, james, checkOutDate);
+
+    CheckInByBarcodeResponse checkInResponse = checkInFixture.checkInByBarcode(
+      new CheckInByBarcodeRequestBuilder()
+        .forItem(nod)
+        .on(checkInDate)
+        .at(checkInServicePointId));
+
+    JsonObject checkedInLoan = checkInResponse.getLoan();
+
+    List<JsonObject> publishedEvents = Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(FakePubSub::getPublishedEvents, hasSize(2));
```

Comment:
Why are there 2 events published?

Reply:
First event is for check-out, second is for check-in

Reply:
Could a comment be added to explain this or the events be cleared prior to the check in?

Reply:
Added